### PR TITLE
rewrote big parts

### PR DESCRIPTION
**Pull request by DrLippe**
You might want to double check this. I hope google did a good job on translating the french localization. 
Man, this was a pain! Hope it works out for you too!

### CODE REVIEW

**Comment by quisquous** [comment]
Ohmygosh, this is a lot! Thanks! This might take me a little bit of time to look at and test locally, just as a warning.

File: ui/raidboss/data/triggers/o9s.js [hunk 1]
```diff
@@ -182,16 +151,365 @@
...
+        data.entropyCount = data.entropyCount || 0;
+      },
+      delaySeconds: function(data) {
+        if ( (data.role == 'tank' || data.role == 'healer') && data.phaseType == 'fire' )
+          return 19;
+        if ( (data.role != 'tank' && data.role != 'healer') && data.phaseType == 'fire' )
+          return 5;
+      },
+      infoText: function(data) {
+        if ( data.phaseType == 'fire' ) {
```
Comment (quisquous) [inline ui/raidboss/data/triggers/o9s.js]: Thanks for this too.  I was going to look at fixing the "Hide middle" during orb phase, but you've already fixed that!

File: ui/raidboss/data/triggers/o9s.js [hunk 2]
```diff
@@ -182,16 +151,365 @@
...
+    {
+      id: 'O9S dynamic fluid 1',
+      regex: /:(\y{Name}) gains the effect of (?:Unknown_641|Dynamic Fluid) from  for (\y{Float}) Seconds/,
+      regexDe: /:(\y{Name}) gains the effect of (?:Unknown_641|Chaosspritzer) from  for (\y{Float}) Seconds/,
+      regexFr: /:(\y{Name}) gains the effect of (?:Unknown_641|Eaux du chaos) from  for (\y{Float}) Seconds/,
+      regexJa: /:(\y{Name}) gains the effect of (?:Unknown_641|混沌の水) from  for (\y{Float}) Seconds/,
+      condition: function(data, matches) {
+        return matches[1] == data.me;
+      },
+      delaySeconds: function(data, matches) {
```
Comment (quisquous) [inline ui/raidboss/data/triggers/o9s.js]: Should there be a "stack donut" call for both the dps and the T/H ones, and not just the one that is on you?

File: ui/raidboss/data/triggers/o9s.js [hunk 3]
```diff
@@ -200,39 +518,145 @@
       },
     },
     {
-      id: 'O9S Primordial Crust',
+      id: 'O9S primordial crust',
```
Comment (quisquous) [inline ui/raidboss/data/triggers/o9s.js]: Can you capitalize all the ids?

File: ui/raidboss/data/triggers/o9s.js [hunk 4]
```diff
@@ -139,19 +76,41 @@
...
       regexDe: /14:3173:Chaos starts using Horizontale Implosion/,
       regexFr: /14:3173:Chaos starts using Implosion Horizontale/,
       regexJa: /14:3173:カオス starts using ホリゾンタルインプロージョン/,
+      preRun: function(data) {
+        data.primordialCrust = data.primordialCrust || 'false';
+      },
+      alertText: function(data) {
+        if ( data.primordialCrust == 'true' ) {
+          return {
+            en: 'Sides -> Front/Back',
```
[the remark this answers is not in the document]
Comment (DrLippe) [reply ui/raidboss/data/triggers/o9s.js]: not played as dps so far but you get another callout to die on next mechanic when you gain the effect of primordial crust. so it should print 
Die on next mechanic!
Sides -> Front/Back

File: ui/raidboss/data/triggers/o9s.js [hunk 5]
```diff
@@ -182,16 +151,365 @@
...
+        if ( data.role == 'tank' ) {
+          return {
+            en: 'Orb Tethers',
+            de: 'Kugel-Verbindungen',
+            fr: 'Récupérez l\'orbe',
+          };
+        }
+      },
+      infoText: function(data) {
+        if ( data.role == 'healer' ) {
```
[the remark this answers is not in the document]
Comment (DrLippe) [reply ui/raidboss/data/triggers/o9s.js]: always! especially as tanks go down 90% in their HP these days

File: ui/raidboss/data/triggers/o9s.js [hunk 6]
```diff
@@ -182,16 +151,365 @@
...
+      regexJa: /:(\y{Name}) gains the effect of (?:Unknown_640|混沌の炎) from  for (\y{Float}) Seconds/,
+      condition: function(data, matches) {
+        return matches[1] == data.me;
+      },
+      preRun: function(data) {
+        data.entropyCount = data.entropyCount || 0;
+        if ( data.phaseType == 'fire' )
+          data.entropyCount += 1;
+      },
+      delaySeconds: function(data, matches) {
```
[the remark this answers is not in the document]
Comment (DrLippe) [reply ui/raidboss/data/triggers/o9s.js]: the delay seconds one? no, dps get a callout for stacking prior to their 2nd entropy.

File: ui/raidboss/data/triggers/o9s.js (hunk 2 above)
Comment (DrLippe) [reply ui/raidboss/data/triggers/o9s.js]: Yes, thats the whole point of having 2 dynamic fluid triggers. If there is the possibility to have 2 delays in one trigger I could fuse them.

**Comment by DrLippe** [comment]
Just found out that the enrage phase tracking doesnt work out well. Trying to fix that. 

Would be a lot easier if you could emulate this fight somehow (maybe feature request?) ^^"

File: ui/raidboss/data/triggers/o9s.js [hunk 7]
```diff
@@ -139,19 +76,41 @@
         };
       },
       tts: function(data) {
+        if ( data.primordialCrust == 'true' ) {
```
[the remark this answers is not in the document]
Comment (DrLippe) [reply ui/raidboss/data/triggers/o9s.js]: I'm a total noob when it comes to code. It's mostly trial and error for me, I got no clue how to use npm but I just did it with search and replace 👍

**Comment by quisquous** [comment]
Thanks for this! I tested this locally with a few other changes and it looks good.  I'm out of town this weekend, but I'll try to land this soon.

**Pull request merged**